dsc/src/subcommand.rs

```diff
@@ -320,14 +320,15 @@ pub fn resource(subcommand: &ResourceSubCommand, stdin: &Option<String>) {
         ResourceSubCommand::List { resource_name, description, tags, format } => {
 
             let mut write_table = false;
-            let mut table = Table::new(&["Type", "Version", "Requires", "Description"]);
+            let mut methods = String::new();
+            let mut table = Table::new(&["Type", "Version", "Methods", "Requires", "Description"]);
             if format.is_none() && atty::is(Stream::Stdout) {
-                // write as table if fornat is not specified and interactive
+                // write as table if format is not specified and interactive
                 write_table = true;
             }
             for resource in dsc.list_available_resources(&resource_name.clone().unwrap_or_default()) {
                 // if description is specified, skip if resource description does not contain it
-                if description.is_some() || tags.is_some() {
+                if description.is_some() || tags.is_some() || write_table {
```

Comment:
This doesn't seem right.  The description on line 329 probably should be updated, but this `if` is if the user specified either the `--description` or `--tags` filters then it goes in here to perform the filtering.

Reply:
I added the `write_table` here because it also requires the resource manifest, and I didn't want to duplicate effort to retrieve it.

I initially tried pulling the resource manifest code out of this `if` but the `continue` / `break` logic broke and would require additional refactoring to fix.

I tested `dsc resource list` with and without `--description` and I think it behaves as expected.